microproxy/protocol/tls.py

```diff
@@ -91,9 +93,76 @@ def create_src_sslcontext(cert, priv_key, alpn_callback=None,
     ssl_ctx.use_certificate(cert)
     ssl_ctx.use_privatekey(priv_key)
 
-    if alpn_callback:
+    if alpn_callback and HAS_ALPN:
         ssl_ctx.set_alpn_select_callback(alpn_callback)
     if info_callback:
         ssl_ctx.set_info_callback(info_callback)
 
     return ssl_ctx
+
+
+class ServerConnection(object):
+    def __init__(self, stream):
+        self.stream = stream
+        self.alpn_resolver = None
+        self.on_alpn = None
+
+    def start_tls(self, cert, priv_key, info_callback=None,
+                  on_alpn=None, alpn_resolver=None):
+        self.alpn_resolver = alpn_resolver
+        self.on_alpn = on_alpn
+        ssl_ctx = create_src_sslcontext(
```

Comment:
Just saw the message. I was too tired yesterday and fell asleep when I finished my dinner. XD

I think your concern is correct. The original function has done too many things that may be difficult to understand. Separate into two functions is okay for me.

I'll continue reviewing the code, thanks.